build-tools/packages/build-cli/src/library/repoPolicyCheck/fluidBuildTasks.ts

```diff
@@ -351,6 +357,17 @@ function hasTaskDependency(
 	const rootConfig = loadFluidBuildConfig(root);
 	const globalTaskDefinitions = normalizeGlobalTaskDefinitions(rootConfig?.tasks);
 	const taskDefinitions = getTaskDefinitions(json, globalTaskDefinitions, false);
+	// Searched deps that are package specific (e.g. <packageName>#<taskName>)
+	// It is expected that all packageNames are other packages' names; using
+	// given package's name (json.name) will alway return false as package is
+	// not a dependency of itself. Skip "name# prefix for self dependencies.
+	const packageSpecificSearchDeps = searchDeps.filter((d) => d.includes("#"));
+	// Set of package dependencies
+	const packageDependencies = new Set([
```

Comment:
I know we don't currently leverage them, but should this theoretically include `peerDependencies` as well?

Reply:
Yes, and I think optional peer deps too. This is written to match the task build-tools buildGraph logic that uses Package.combinedDependencies. I'll add a note.